app/models/mixins/per_ems_worker_mixin.rb

```diff
@@ -18,7 +18,7 @@ def all_ems_in_zone
     end
 
     def all_valid_ems_in_zone
-      all_ems_in_zone.select(&:authentication_status_ok?)
+      all_ems_in_zone.select { |ems| ems.authentication_status_ok? && ems.enabled? }
```

Comment:
Put enabled first as it's a column and doesn't require hitting an association (minor performance)